lib/src/components/NxFileUpload/NxMultiFileUpload/__tests__/NxMultiFileUpload.test.tsx

```diff
@@ -0,0 +1,125 @@
+import { pipe } from 'ramda';
+
+import { rtlRender, rtlRenderElement } from '../../../../__testutils__/rtlUtils';
+import NxMultiFileUpload from '../NxMultiFileUpload';
+import NxForm from '../../../NxForm/NxForm';
+
+describe('NxMultiFileUpload', function() {
+  const minimalProps = {
+        files: null,
+        onChange: () => {}
+      },
+      quickRender = rtlRender(NxMultiFileUpload, minimalProps),
+      renderEl = rtlRenderElement(NxMultiFileUpload, minimalProps),
+      renderInput = pipe(renderEl, el => el?.querySelector('input[type=file]'));
+
+  it('renders a file input and a button', function() {
+    const component = quickRender(),
+
+        // file inputs have no aria role and the label text is outside of the scope of this component,
+        // so we have to query like this
+        input = component.container.querySelector('input[type=file]'),
+        button = component.queryByRole('button', { name: 'Choose File' });
+
+    expect(input).toBeDefined();
+    expect(button).toBeDefined();
+
+    // cannot effectively test that the button activates the input here, will do in functional test
+  });
+
+  it('sets specified ClassNames on the top-level element', function() {
+    const component = renderEl()!;
+    const customComponent = renderEl({ className: 'foo'});
+
+    expect(customComponent).toHaveClass('foo');
+
+    for (const cls of Array.from(component.classList)) {
+      expect(customComponent).toHaveClass(cls);
+    }
+  });
+
+  it('sets specified attributes on the input', function() {
+    const component = renderEl({id: 'bar', lang: 'en' })!,
+        input = component.querySelector('input[type=file]');
+
+    expect(input).toHaveAttribute('id', 'bar');
+    expect(input).toHaveAttribute('lang', 'en');
+  });
+
+  it('sets the multiple attribute on the input', function() {
+    const component = renderEl()!,
+        input = component.querySelector('input[type=file]');
+
+    expect(input).toHaveAttribute('multiple');
+  });
+
+  it('renders a "No file selected" message if files is null', function() {
+    expect(renderEl()).toHaveTextContent('No file selected');
+  });
+
+  it('renders a "No file selected" aria description if files is null', function() {
+    expect(renderInput()).toHaveAccessibleDescription('No file selected');
+  });
+
+  it('shows an error when there is no file selected, if it isRequired is not isPristine', function() {
+    expect(quickRender().queryByRole('alert')).not.toBeTruthy();
+    expect(quickRender({ isPristine: true }).queryByRole('alert')).not.toBeTruthy();
+    expect(quickRender({ isRequired: true, isPristine: true }).queryByRole('alert')).not.toBeTruthy();
+
+    expect(quickRender({ isRequired: true, isPristine: false }).queryByRole('alert'))
+        .toHaveTextContent('This field is required!');
+  });
+
+  it('shows an error when in a form with showValidationErrors if it is isRequired and no file is selected', function() {
```

Comment:
This should test that this happens whether or not `isPristine` is true

Reply:
> This should test that this happens whether or not `isPristine` is true

added in testing here: [7aaf779](https://github.com/sonatype/sonatype-react-shared-components/pull/850/commits/7aaf77915e4b2336966bd5aa6116ea9b96c7ce18)